### PR TITLE
`Exam mode`: Fix an issue where individual working time is not displayed in sidebar

### LINKED ISSUE #8917: `Exam mode`: Individual working time not displayed in sidebar
### Describe the bug

The individual working time of a student is not reflected in the sidebar.

### To Reproduce

1. Create an exam and give a student an individual working time
2. See that the working time is displayed properly in the exam preview screen
3. See that the working time is not displayed properly in the sidebar

### Expected behavior

The working time in the sidebar should reflect the individual working time of the student.

### Screenshots

image

### Which version of Artemis are you seeing the problem on?

7.3.1

### What browsers are you seeing the problem on?

Chrome

### Additional context

Found by @anian03 during exam testing session

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by edkaya**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context

It fixes #8917.

### Description

We now load the student exam of each real exam in course exams module to get individual working time for the student. 

### Steps for Testing

Prerequisites:
- 1 Exam
- 1 registered student with individual working time

1. Log in to Artemis
2. Navigate to Course Administration
3. Create an exam 
4. Click Students 
5. Register the student
6. Click Student Exams 
7. Click Generate individual exams 
8. Click View 
9. Set additional working time and click save 
10. Click Prepare exercise start
11. Log in as registered student
12. Go to Exams
13. Verify that sidebar has the individual working time like the information box in welcome page of the exam

#### Exam Mode Testing

Prerequisites:
- 1 Exam
- 1 Student without an individual working time

1. Log in to Artemis
2. Go to exams
3. Make sure that sidebar has correct working time if there is no individual working time is set for the student

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2
#### Exam Mode Test
- [x] Test 1
- [x] Test 2

### Screenshots

Before 
Screenshot 2024-07-12 at 02 12 29

After
Screenshot 2024-07-12 at 01 44 43

## Summary by CodeRabbit

- **New Features**
  - Enhanced functionality in managing student exams related to real exams in the course overview.
  - Added the ability to efficiently retrieve and map student exams for real exams.

### CODE REVIEW

**Review by coolchock — DISMISSED** [review]
tested on ts4, works as expected. Code LGTM

**Review by JohannesStoehr — CHANGES_REQUESTED** [review]
Code looks good overall, left two nitpicks:

**Review by egekurt123 — APPROVED** [review]
Demonstrated in a testing session on ts6, works as expected

**Review by EneaGore — APPROVED** [review]
Demonstrated  on testing session. Works as expected.

File: src/main/webapp/app/overview/course-exams/course-exams.component.ts [hunk 1]
```diff
@@ -249,4 +261,12 @@ export class CourseExamsComponent implements OnInit, OnDestroy {
         }
         this.navigateToExam();
     }
+
+    getAllStudentExamsForRealExams(): StudentExam[] {
+        const allStudentExams: StudentExam[] = [];
+        this.studentExamsForRealExams.forEach((studentExam) => {
+            allStudentExams.push(studentExam);
+        });
+        return allStudentExams;
```
[the remark this answers is not in the document]
Comment (JohannesStoehr) [reply src/main/webapp/app/overview/course-exams/course-exams.component.ts]: That's also fine!

**Review by JohannesStoehr — APPROVED** [review]
Code

**Review by pzdr7 — APPROVED** [review]
Code 👍

**Pull request merged**